src/main/java/org/mybatis/spring/annotation/MapperScannerRegistrar.java

```diff
@@ -119,6 +119,10 @@ void registerBeanDefinitions(AnnotationAttributes annoAttrs, BeanDefinitionRegis
     basePackages.addAll(Arrays.stream(annoAttrs.getClassArray("basePackageClasses")).map(ClassUtils::getPackageName)
         .collect(Collectors.toList()));
 
+    if (basePackages.isEmpty()) {
+      basePackages.add(defaultBasePackage);
```

Comment:
I prefer to call the `getDefaultBasePackage()`  in this place.

Reply:
the declared class name is in `AnnotationMetadata` and `registerBeanDefinitions` method is only able to access `AnnotationAttributes` which does not have the class name. Do you prefer change the method signature to pass  `AnnotationMetadata` ?

Reply:
Yes.

Reply:
Added!